feat: login page

Dudleyneedham opened the pull request:
## fixes https://github.com/KILTprotocol/ticket/issues/898

- User gets password prompt
- password is salted and hashed
- result is used to decrypt data
- if not successful -> password is wrong
- use same result to encrypt future data
- error handling
- styling

## How to test:

Spin up the demo-client and try to enter the site with correct and incorrect passwor.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

Dudleyneedham commented:
![Screenshot 2021-01-21 at 13 32 15](https://user-images.githubusercontent.com/48058984/105351652-1ef9db80-5bed-11eb-8312-c52c047c02bf.png)

tjwelde commented on src/containers/LoginGate/LoginGate.scss:
```diff
@@ -0,0 +1,53 @@
+    }
+  }
+
+  & form {
```
You should only style what is actually inside this component (LoginGate), not the other components, which are included. _Separation of Concerns_

tjwelde commented:
Closd button is not aligned
<img width="184" alt="Screenshot 2021-01-26 at 14 11 15" src="https://user-images.githubusercontent.com/1248214/105849273-6fec4400-5fe0-11eb-9b08-0f9bb7d5e69d.png">

tjwelde commented:
"password" is written wrong and could use a question mark at the end.
<img width="671" alt="Screenshot 2021-01-26 at 14 12 31" src="https://user-images.githubusercontent.com/1248214/105849348-8db9a900-5fe0-11eb-80ae-90d6a6471346.png">

tjwelde commented:
Lots of different paddings. Please make it more consistent
<img width="469" alt="Screenshot 2021-01-26 at 14 13 36" src="https://user-images.githubusercontent.com/1248214/105849457-b6da3980-5fe0-11eb-89e3-6fddc17e67c1.png">

tjwelde commented:
Form doesn't allow 12 character passwords, besides it saying that you should use 12 characters or more

tjwelde commented on src/components/Register/Register.tsx:
```diff
@@ -0,0 +1,65 @@
+      setErrorMessage('Please, enter a password')
+      return setError(true)
+    }
+    if (password.length <= 11) {
```
Why not just
```suggestion
    if (password.length < 12) {
```
?

tjwelde commented on src/components/Register/Register.tsx:
```diff
@@ -0,0 +1,65 @@
+            </div>
+          </label>
+          <button type="submit" className="confirm">
+            Login
```
Rename to "Register", or "Create Account"

Dudleyneedham replied:
Changed to "Create Account".

tjwelde reviewed (changes requested):
The input and the button are not horizontally aligned (both login and register)
![Screenshot 2021-01-26 at 18 23 45](https://user-images.githubusercontent.com/1248214/105880664-e9e1f480-6003-11eb-8bee-5abda7da5e61.png)

The close button is still not aligned well. it has more margin on bottom than on the top. Should be in the middle vertically.
![Screenshot 2021-01-26 at 18 25 05](https://user-images.githubusercontent.com/1248214/105880747-02520f00-6004-11eb-8aa8-68de22b92002.png)

Dudleyneedham commented:
> The input and the button are not horizontally aligned (both login and register)
> ![Screenshot 2021-01-26 at 18 23 45](https://user-images.githubusercontent.com/1248214/105880664- e9e1f480-6003-11eb-8bee-5abda7da5e61.png)
> 
> The close button is still not aligned well. it has more margin on bottom than on the top. Should be in the middle vertically.
> ![Screenshot 2021-01-26 at 18 25 05](https://user-images.githubusercontent.com/1248214/105880747-02520f00-6004-11eb-8aa8-68de22b92002.png)

I have updated it through-out the whole application. I know it isn't part of the ticket, but the issue is from the `Modal.scss` so I fixed it there.

Let me know how you find it? 
<img width="353" alt="Screenshot 2021-01-27 at 10 40 08" src="https://user-images.githubusercontent.com/48058984/105972561-07f33780-608c-11eb-8656-fbc7667011b0.png">

tjwelde commented:
I like it

tjwelde reviewed (approved):
LGTM

![](https://media2.giphy.com/media/fQoxwZBVWq5jhLXRty/giphy.gif)
